pennylane/compiler/qjit_api.py

```diff
@@ -537,6 +541,37 @@ def _decorator(body_fn):
     return _decorator
 
 
+@functools.lru_cache
+def _get_for_loop_qfunc_prim():
+    """Get the loop_for primitive for quantum functions."""
+
+    import jax  # pylint: disable=import-outside-toplevel
```

Comment:
Importing `jax` here wouldn't add an overhead? 🤔

Reply:
We must use `jax` functions within `_get_for_loop_qfunc_prim`. We used this scheme for all the previous implementations, but (although I am not 100% sure) I think the overhead does almost not exist in this case since when we reach this part of the code `jax` should have been imported already

Reply:
We can't import jax outside this function, since that would cause `import pennylane as qml` to fail if jax is not installed.  We have to hide the dependency and make the jax import a bit lazy. This primitive is only defined "as needed", instead of "on import".

But we don't want to define the primitive a bunch of times with additional overhead, so we slapped the `lru_cache` around this constructor.

That makes it so the primitive is created once the first time it's needed, and then just retrieved every other time.